config/routes.rb

```diff
@@ -0,0 +1,34 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  
+  root :to => "tasks#index"
+  
+  resources :tasks, only: [:index, :show, :new, :create, :edit, :update]
+  # resources :tasks, except: [:name, :description]
+  
+# # Routes that operate on the task collection
+#   get '/tasks', to: 'tasks#index', as: 'tasks'
+# # Step 1: Create a view so that a user sees the New task Form
+#   get '/tasks/new', to: 'tasks#new', as: 'new_task' 
+# # Step 2: Create an action so that the form data actually gets processed by Rails (by the server) and creates that new task, and changes the database
+#   post '/tasks', to: 'tasks#create'
+
+# # Routes that operate on individual tasks
+#   get '/tasks/:id', to: 'tasks#show', as: "task"
+
+#   get '/tasks/:id/edit', to: 'tasks#edit', as: 'edit_task'
+
+  #  patch '/tasks/update/:id', to: 'tasks#update', as: 'task_update'
+
+  delete '/tasks/:id', to: 'tasks#destroy', as: 'task_delete'
+
+  get '/tasks/complete/:id', to: 'tasks#toggle_complete', as: 'complete_task'
```

Comment:
What is the controller action `tasks#toggle_complete` (TasksController, toggle_complete action) supposed to do? It's supposed to change a task's data so that it's complete. Since it's going to change task data, instead of ask/query for data, we should think about the HTTP Verb for this route. The HTTP Verb you defined is `get`, which is usually associated with asking/querying for information. Since we're not querying, we should use the HTTP Verbs associated with changing information, like `patch` or `put`. Therefore, maybe the better version of this route may be `patch '/tasks/complete/:id', to: 'tasks#toggle_complete', as: 'complete_task'`

Of course, this will affect the link in the view that goes to this route.